File: Components/WinForms/CS/GridMainDemo/Modules/ExcelFiltering.cs
using DevExpress.XtraGrid.Views.Base;
using DevExpress.Utils.Menu;
using DevExpress.Data.Filtering;

namespace DevExpress.XtraGrid.Demos {
    public partial class ExcelFiltering : TutorialControl {
        public override BaseView ExportView {
            get { return bandedGridView1; }
        }
        protected override void InitMDBData(string connectionString) {
            VehiclesData.InitOrdersData(connectionString, gridControl1, 10000, 400);
            VehiclesData.InitColumnViewEditors(bandedGridView1);
        }
        public ExcelFiltering() {
            CreateWaitDialog();
            InitializeComponent();
            InitVehiclesData();
            bandedGridView1.ActiveFilterCriteria = new GroupOperator(GroupOperatorType.Or,  
                new FunctionOperator(FunctionOperatorType.IsOutlookIntervalYesterday, new OperandProperty("SalesDate")),
                new FunctionOperator(FunctionOperatorType.IsOutlookIntervalToday, new OperandProperty("SalesDate")));
        }
        #region GenerateReport
        public override bool AllowGenerateReport { get { return false; } }
        #endregion
    }
}

File: Components/WinForms/VB/GridMainDemo/Modules/ExcelFiltering.vb
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.Utils.Menu
Imports DevExpress.Data.Filtering

Namespace DevExpress.XtraGrid.Demos
    Partial Public Class ExcelFiltering
        Inherits TutorialControl

        Public Overrides ReadOnly Property ExportView() As BaseView
            Get
                Return bandedGridView1
            End Get
        End Property
        Protected Overrides Sub InitMDBData(ByVal connectionString As String)
            VehiclesData.InitOrdersData(connectionString, gridControl1, 10000, 400)
            VehiclesData.InitColumnViewEditors(bandedGridView1)
        End Sub
        Public Sub New()
            CreateWaitDialog()
            InitializeComponent()
            InitVehiclesData()
            bandedGridView1.ActiveFilterCriteria = New GroupOperator(GroupOperatorType.Or, _
                New FunctionOperator(FunctionOperatorType.IsOutlookIntervalYesterday, New OperandProperty("SalesDate")), _
                New FunctionOperator(FunctionOperatorType.IsOutlookIntervalToday, New OperandProperty("SalesDate")))
        End Sub
#Region "GenerateReport"
        Public Overrides ReadOnly Property AllowGenerateReport() As Boolean
            Get
                Return False
            End Get
        End Property
#End Region
    End Class
End Namespace
